File: CPP/Clothing.cpp
/* Saya Alfen Fajri Nurulhaq (2201431) mengerjakan Latihan Prakitum 2 DPBO Inheritance dalam mata
   kuliah Strukdat untuk keberkahanNya maka saya tidak
   melakukan kecurangan seperti yang telah dispesikasikan.Aamiin*/
#include "Product.cpp"  //Product.cpp sebagai kelas induk (parent class)

#include <iostream>     //library IO untuk menggunakan fungsi input/output standar
#include <string>       //library string untuk menggunakan tipe data string

using namespace std;

class Clothing : public Product  // Tentukan kelas Clothing sebagai subkelas (derived class) dari kelas Product
{
private:  // Definisikan properti-properti privat
    string size;      // Properti untuk menyimpan ukuran pakaian
    string material;  // Properti untuk menyimpan bahan pakaian
    string gender;    // Properti untuk menyimpan pakaian laki laki dan perempuan

public:
    // Method-Method publik
    // Konstruktor
    Clothing() : Product()  // Konstruktor default untuk Clothing, meng-extend ke kelas Product
    {
        this->size = "";
        this->material = "";
        this->gender = "";
    }

    Clothing(int idProduct, string name, string brand, int price, string size, string material, string gender)
        : Product(idProduct, name, brand, price)  // Konstruktor dengan parameter, memanggil konstruktor kelas Product menggunakan initializer list
    {
        this->size = size;
        this->material = material;
        this->gender = gender;
    }

    // Method getter dan setter untuk size
    string getSize()
    {
        return this->size;
    }

    void setSize(string size)
    {
        this->size = size;
    }

    // Method getter dan setter untuk material
    string getMaterial()
    {
        return this->material;
    }

    void setMaterial(string material)
    {
        this->material = material;
    }

    // Method getter dan setter untuk gender
    string getGender()
    {
        return this->gender;
    }

    void setGender(string gender)
    {
        this->gender = gender;
    }

    // Destruktor
    ~Clothing()
    {
        // Destruktor kosong dalam kasus ini
    }
};

File: PHP LATPRAK 2/Clothing.php
<?php
/* Saya Alfen Fajri Nurulhaq (2201431) mengerjakan Latihan Prakitum 2 OOP Inheritance dalam mata
   kuliah Strukdat untuk keberkahanNya maka saya tidak
   melakukan kecurangan seperti yang telah dispesikasikan.Aamiin*/
require_once 'Product.php';
// Mendefinisikan kelas Clothing sebagai subclass dari class Product
class Clothing extends Product {
    // Mendeklarasikan variabel privat untuk size, material, dan gender
    private $size = "";
    private $material = "";
    private $gender = "";

    // Konstruktor dengan parameter untuk inisialisasi nilai awal
    public function __construct($idProduct, $name, $brand, $price, $size = "", $material = "", $gender = "") {
        // Memanggil konstruktor dari kelas induk (Product) dengan menggunakan parent::__construct()
        parent::__construct($idProduct, $name, $brand, $price);
        // Menginisialisasi variabel dengan nilai dari parameter konstruktor
        $this->size = $size;
        $this->material = $material;
        $this->gender = $gender;
    }

    // Metode untuk mendapatkan nilai size
    public function getSize() {
        return $this->size;
    }

    // Metode untuk mengatur nilai size
    public function setSize($size) {
        $this->size = $size;
    }

    // Metode untuk mendapatkan nilai material
    public function getMaterial() {
        return $this->material;
    }

    // Metode untuk mengatur nilai material
    public function setMaterial($material) {
        $this->material = $material;
    }

    // Metode untuk mendapatkan nilai gender
    public function getGender() {
        return $this->gender;
    }

    // Metode untuk mengatur nilai gender
    public function setGender($gender) {
        $this->gender = $gender;
    }

    public function __destruct(){

    }
}
?>
